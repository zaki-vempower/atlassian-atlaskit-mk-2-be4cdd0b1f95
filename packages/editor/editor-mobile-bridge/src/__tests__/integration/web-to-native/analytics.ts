import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  editor,
  renderer,
  skipBrowsers as skip,
  getBridgeOutput,
} from '../_utils';

BrowserTestCase(
  'editor: calls analyticsBridge.trackEvent when analytics events are captured',
  { skip },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editor.placeholder);

    // just loading the editor will fire three events:
    // editor started, editor mounted, editor proseMirrorRendered

    // insert a bullet list to fire a user generated event
    await browser.type(editor.placeholder, '* ');

    // both editor mounted and editor proseMirrorRendered events include timings
    // which will change each time so we exclude them
    const [editorStartedEvent, , , actionInsertedEvent] = await getBridgeOutput(
      browser,
      'analyticsBridge',
      'trackEvent',
    );

    expect([editorStartedEvent, actionInsertedEvent]).toMatchCustomSnapshot(
      testName,
    );
  },
);

BrowserTestCase(
  'renderer: calls analyticsBridge.trackEvent when analytics events are captured',
  { skip },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(renderer.path);
    await browser.waitForSelector(renderer.placeholder);

    // just loading the renderer will fire two events:
    // renderer started, renderer rendered
    // renderer rendered includes timings which will change each time so we exclude it
    const [rendererStartedEvent] = await getBridgeOutput(
      browser,
      'analyticsBridge',
      'trackEvent',
    );

    expect(rendererStartedEvent).toMatchCustomSnapshot(testName);
  },
);
