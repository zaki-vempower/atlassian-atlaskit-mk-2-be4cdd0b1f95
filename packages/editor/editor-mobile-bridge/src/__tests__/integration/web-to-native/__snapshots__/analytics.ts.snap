// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editor: calls analyticsBridge.trackEvent when analytics events are captured 1`] = `
Array [
  Object {
    "event": "{\\"source\\":\\"unknown\\",\\"action\\":\\"started\\",\\"actionSubject\\":\\"editor\\",\\"attributes\\":{\\"packageName\\":\\"@atlaskit/editor-core\\",\\"packageVersion\\":\\"999.9.9\\",\\"componentName\\":\\"editorCore\\",\\"appearance\\":\\"mobile\\",\\"platform\\":\\"web\\"},\\"tags\\":[\\"fabricEditor\\"],\\"eventType\\":\\"ui\\"}",
  },
  Object {
    "event": "{\\"source\\":\\"unknown\\",\\"action\\":\\"formatted\\",\\"actionSubject\\":\\"text\\",\\"actionSubjectId\\":\\"bulletedList\\",\\"attributes\\":{\\"packageName\\":\\"@atlaskit/editor-core\\",\\"packageVersion\\":\\"999.9.9\\",\\"componentName\\":\\"editorCore\\",\\"appearance\\":\\"mobile\\",\\"inputMethod\\":\\"autoformatting\\"},\\"tags\\":[\\"fabricEditor\\"],\\"eventType\\":\\"track\\"}",
  },
]
`;

exports[`editor: calls analyticsBridge.trackEvent when analytics events are captured 2`] = `
Array [
  Object {
    "event": "{\\"source\\":\\"unknown\\",\\"action\\":\\"started\\",\\"actionSubject\\":\\"editor\\",\\"attributes\\":{\\"packageName\\":\\"@atlaskit/editor-core\\",\\"packageVersion\\":\\"999.9.9\\",\\"componentName\\":\\"editorCore\\",\\"appearance\\":\\"mobile\\",\\"platform\\":\\"web\\"},\\"tags\\":[\\"fabricEditor\\"],\\"eventType\\":\\"ui\\"}",
  },
  Object {
    "event": "{\\"source\\":\\"unknown\\",\\"action\\":\\"formatted\\",\\"actionSubject\\":\\"text\\",\\"actionSubjectId\\":\\"bulletedList\\",\\"attributes\\":{\\"packageName\\":\\"@atlaskit/editor-core\\",\\"packageVersion\\":\\"999.9.9\\",\\"componentName\\":\\"editorCore\\",\\"appearance\\":\\"mobile\\",\\"inputMethod\\":\\"autoformatting\\"},\\"tags\\":[\\"fabricEditor\\"],\\"eventType\\":\\"track\\"}",
  },
]
`;

exports[`renderer: calls analyticsBridge.trackEvent when analytics events are captured 1`] = `
Object {
  "event": "{\\"source\\":\\"unknown\\",\\"action\\":\\"started\\",\\"actionSubject\\":\\"renderer\\",\\"attributes\\":{\\"appearance\\":\\"mobile\\",\\"packageName\\":\\"@atlaskit/renderer\\",\\"packageVersion\\":\\"999.9.9\\",\\"componentName\\":\\"editorCore\\",\\"platform\\":\\"web\\"},\\"tags\\":[\\"fabricEditor\\"],\\"eventType\\":\\"ui\\"}",
}
`;

exports[`renderer: calls analyticsBridge.trackEvent when analytics events are captured 2`] = `
Object {
  "event": "{\\"source\\":\\"unknown\\",\\"action\\":\\"started\\",\\"actionSubject\\":\\"renderer\\",\\"attributes\\":{\\"appearance\\":\\"mobile\\",\\"packageName\\":\\"@atlaskit/renderer\\",\\"packageVersion\\":\\"999.9.9\\",\\"componentName\\":\\"editorCore\\",\\"platform\\":\\"web\\"},\\"tags\\":[\\"fabricEditor\\"],\\"eventType\\":\\"ui\\"}",
}
`;
